launch/launch/launch_service.py

```diff
@@ -185,37 +184,28 @@ def _prepare_run_loop(self):
                 this_task = asyncio.Task.current_task(this_loop)
 
             self.__this_task = this_task
-            # Setup custom signal handlers for SIGINT, SIGTERM and maybe SIGQUIT.
-            sigint_received = False
-
-            def _on_sigint(signum):
-                nonlocal sigint_received
-                base_msg = 'user interrupted with ctrl-c (SIGINT)'
-                if not sigint_received:
+            # Setup custom signal handlers for SIGINT & SIGTERM
+            signals_received = set()
+
+            def _on_signal(signum):
+                nonlocal signals_received
+                signal_name = signal.Signals(signum).name
+                base_msg = 'caught ' + signal_name
+                if signal_name not in signals_received:
+                    signals_received.add(signal_name)
+                    due_to_sigint = True if signal_name == 'SIGINT' else False
                     self.__logger.warning(base_msg)
                     ret = self._shutdown(
-                        reason='ctrl-c (SIGINT)', due_to_sigint=True, force_sync=True
+                        reason=base_msg, due_to_sigint=due_to_sigint, force_sync=True
```

Comment:
@ciandonovan meta: sending a shutown event is not the same as pulling the brakes on the event loop, but I can see how the previous implementation can cause problems when managing `launch` itself. Most process managers use SIGTERM. I guess you could argue that SIGKILL escalation is appropriate if a shutdown isn't enough.